src/App/ComplexGeoData.cpp

```diff
@@ -55,17 +1218,15 @@ ComplexGeoData::~ComplexGeoData()
 
 Data::Segment* ComplexGeoData::getSubElementByName(const char* name) const
 {
-    int index = 0;
-    std::string element;
-    boost::regex ex("^([^0-9]*)([0-9]*)$");
-    boost::cmatch what;
-
-    if (boost::regex_match(name, what, ex)) {
-        element = what[1].str();
-        index = std::atoi(what[2].str().c_str());
-    }
-
-    return getSubElement(element.c_str(), static_cast<unsigned long>(index));
+    int index = 0;
+    std::string element(name);
+    std::string::size_type pos = element.find_first_of("0123456789");
+    if (pos != std::string::npos) {
+        index = std::atoi(element.substr(pos).c_str());
+        element = element.substr(0,pos);
+    }
+
+    return getSubElement(element.c_str(),index);
 }

```

Comment:
Is it necessary to remove the regular expression here?
Some months ago I added the use of regex in some parts of the code to avoid using wrong names for subelements.